drivers/bus/isapnp/hardware.c

```diff
@@ -1,21 +1,33 @@
 /*
  * PROJECT:         ReactOS ISA PnP Bus driver
- * FILE:            hardware.c
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
  * PURPOSE:         Hardware support code
- * PROGRAMMERS:     Cameron Gutman (cameron.gutman@reactos.org)
- *                  Hervé Poussineau
+ * COPYRIGHT:       Copyright 2010 Cameron Gutman (cameron.gutman@reactos.org)
+ *                  Copyright 2020 Hervé Poussineau (hpoussin@reactos.org)
+ *                  Copyright 2021 Dmitry Borisov (di.sean@protonmail.com)
  */
 
-#include <isapnp.h>
+#include "isapnp.h"
 
 #define NDEBUG
 #include <debug.h>
 
+#ifdef _MSC_VER
+#pragma warning(disable:28138) /* Legacy bus */
```

Comment:
What does that do? (regarding "Legacy bus")

Reply:
The VS2019 code analyzer with "AllRules" rule selected reports this [warning](https://docs.microsoft.com/en-us/cpp/code-quality/c28138?view=msvc-160). ISA bus always uses hardcoded values in port addresses.

Reply:
Ok then i would advise putting your last sentence as a comment in the code. Otherwise no-one would still know what happens.